tests/test_minimal/test_metadata_schema.py

```diff
@@ -0,0 +1,201 @@
+            SegmentationImages=dict(
+                name="SegmentationImages",
+                PlaneSegmentationChan1Plane0=dict(
+                    raw=dict(name="PlaneSegmentationChan1Plane0", description="raw segmentation image"),
+                    neuropil=dict(name="PlaneSegmentationChan1Plane0", description="neuropil segmentation image"),
+                ),
+            ),
+        ),
+    )
+
+    validate_metadata(metadata=metadata, schema=metadata_schema)
+
+
+device_error = "'Devices' is a required property"
+invalid_plane_name = "plane_segmentation_chan1_plane0"
+name_error = "'name' is a required property"
+description_error = "'description' is a required property"
+plane_name_error = f"'{invalid_plane_name}' does not match any of the regexes"
+not_enough_properties_error = "does not have enough properties"
+
+
+def test_invalid_ophys_metadata():
+    metadata_schema = load_dict_from_file(
+        Path(__file__).parent.parent.parent / "src" / "neuroconv" / "schemas" / "metadata_schema.json"
+    )
+
+    def create_valid_plane_name(name: str, number=0):
+        return f'{"".join(map(lambda str: str.capitalize(), name.split("_")))[:-1]}{str(number)}'
+
+    metadata = dict(
+        NWBFile=dict(
+            session_start_time="2020-01-01T00:00:00",
+            session_description="Auto-generated by neuroconv",
+            identifier="1234",
+        ),
+        Ophys=dict(
+            Fluorescence={
+                "name": "Fluorescence",
+                invalid_plane_name: dict(),  # Unchecked
+                create_valid_plane_name(invalid_plane_name): dict(),
+                create_valid_plane_name(invalid_plane_name, 1): dict(
+                    raw=dict(),
+                ),
+            },
+            DFOverF={
+                "name": "DfOverF",
+                invalid_plane_name: dict(),  # Unchecked
+                create_valid_plane_name(invalid_plane_name): dict(),
+                create_valid_plane_name(invalid_plane_name, 1): dict(
+                    raw=dict(),
+                ),
+            },
+            SegmentationImages={
+                "name": "SegmentationImages",
+                invalid_plane_name: dict(),  # Unchecked
+                create_valid_plane_name(invalid_plane_name): dict(),
+                create_valid_plane_name(invalid_plane_name, 1): dict(
+                    raw=dict(),
+                ),
+            },
+        ),
+    )
+
+    validator = jsonschema.Draft7Validator(metadata_schema)
+
+    errors = list(map(lambda e: str(e).split("\n")[0], validator.iter_errors(metadata)))
+
+    iterable = iter(errors)
+    nExpectedErrors = 12
+    assert len(errors) == nExpectedErrors
+
+    assert device_error == next(iterable, None)
+
+    # Fluorescence
+    assert not_enough_properties_error in next(iterable, "")
+    assert name_error == next(iterable, None)
+    assert description_error == next(iterable, None)
+    assert plane_name_error in next(iterable, "")
```

Comment:
Something like the following is the go-to suggestion for making the assertion cleaner and less brittle

Though the repeated error headers across

```suggestion
    expected_errors = [
        "does not have enough properties",
        "does not have enough properties",
        "does not have enough properties",
        "'Devices' is a required property"
        "plane_segmentation_chan1_plane0"
        "'name' is a required property"
        "'description' is a required property"
        "'plane_segmentation_chan1_plane0' does not match any of the regexes"
    ]
    assert sort(errors) == sort(expected_errors)
```

or similarly if you only wanted to check that a handful of the really important expected ones were caught but don't care about the full values

```python
for expected_error in expected_errors:
    assert expected_error in errors
```

Reply:
Updated to match both the `message` and the `path` of each error. Might be too verbose, or there might be a better way to do this joint search in Python / for our testing suite—so just let me know

Reply:
matching more is better than matching less